functions/common/aws-s3-common/src/main/java/org/springframework/cloud/fn/common/aws/s3/AmazonS3Properties.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2020-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.fn.common.aws.s3;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+
+/**
+ * @author Timo Salm
+ */
+@ConfigurationProperties("s3.common")
+public class AmazonS3Properties {
+
+	/**
+	 * Optional S3 endpoint url to connect to s3 compatible storage.
```

Comment:
Let's remove the first `S3` - sounds like tautology. 
More over it probably going to mislead: an `s3://` is really about a resource, not an endpoint connection. This one probably is an `http://` in most cases.